api/src/main/java/org/openmrs/validator/AllergyValidator.java

```diff
@@ -64,6 +64,11 @@ public void validate(Object target, Errors errors) {
 		
 		Allergy allergy = (Allergy) target;
 		
+		    // Additional validation: Ensure allergen does not contain numeric values
+			if (allergy.getAllergen() != null && StringUtils.isNumeric(allergy.getAllergen().getNonCodedAllergen())) {
```

Comment:
avoid using null instead do something llike
`if (StringUtils.isNotEmpty(allergy.getAllergen() ) && StringUtils.isNumeric(allergy.getAllergen().getNonCodedAllergen()))
`

Reply:
Using null is easy to read. But when I am using `isNotEmpty`, I am getting a red line error.

Reply:
`if (StringUtils.isNumeric(Objects.requireNonNull(allergy.getAllergen(), "Allergen must not be null").getNonCodedAllergen()))`
Is this is ok?